gui/src/ruler.hh

```diff
@@ -0,0 +1,189 @@
+#pragma once
+
+#include <gtk/gtk.h>
+#include <boost/smart_ptr.hpp>
```

Comment:
Everywhere else, we include `boost/shared_ptr.hpp`. Not sure if there's a difference. However, in time we'll migrate from boost shared pointers to the standard library ones, and then a naive find-and-replace will miss this `#include`